src/python/pants/backend/python/lint/pyupgrade/subsystem.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from typing import cast
+
+from pants.backend.python.goals.lockfile import PythonLockfileRequest, PythonToolLockfileSentinel
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.custom_types import shell_str
+from pants.util.docutil import git_url
+
+
+class PyUpgrade(PythonToolBase):
+    options_scope = "pyupgrade"
+    help = (
+        "Upgrade syntax for newer versions of the language (https://github.com/asottile/pyupgrade)."
+    )
+
+    default_version = "pyupgrade==2.29.0"
```

Comment:
```suggestion
    default_version = "pyupgrade>=2.29.0,<2.30"
```

(Will require regenerating the lockfile)